src/api/nlopt-in.hpp

```diff
@@ -34,6 +34,8 @@
 #include <cstdlib>
 #include <cstring>
 #include <cmath>
+#include <utility> // std::move
+#include <functional> // std::function
```

Comment:
Note that the [current build scripts don't require C++11](https://github.com/stevengj/nlopt/blob/17826b8dd812c1b8cc10042432e60d06ea35e82d/CMakeLists.txt#L162-L167), which is needed for `std::function`.

We'll either need to change the build to require C++11, or add some magic to disable the `std::function` stuff when `nlopt.hpp` is generated from `nlopt-in.hpp` if C++11 is not found.

Reply:
True, I overlooked this. My feeling would be to require C++11 (after all, we're in 2022, eleven (!) years after it was released), but since this is your repository, I defer this decision to you (: If you say drop `std::function`, I could rewrite the functor functionality using templates I believe.

Reply:
@jschueller @stevengj 
OK, `CMakeLists.txt` script now requires C++11 for C++11 API.